File: dotnet_backend/backend/DTO/GalleryCategoryDto.cs
namespace ComputerSeekho.DTO;

/// <summary>
/// One theme on the Campus Life page: its name, how many photos it holds,
/// and one of them to show as the cover.
///
/// Not a table — it is computed from gallery_images, which is why it has no
/// entity and no AutoMapper profile.
/// </summary>
public record GalleryCategoryDto(
    string Category,
    int ImageCount,
    string? CoverImageUrl);

File: backend/src/main/java/com/smvita/computerseekho/dto/GalleryCategoryDto.java
package com.smvita.computerseekho.dto;

/**
 * One clickable theme on the Campus Life page — "Lab Sessions",
 * "Guest Lectures" and so on.
 *
 * Derived from gallery_images.category rather than stored in its own
 * table: a theme has no attributes of its own beyond its name, so a
 * lookup table would add a join and a maintenance screen without adding
 * any information.
 */
public record GalleryCategoryDto(
        String category,
        String coverImageUrl,
        int photoCount
) {}
